Userland/Libraries/LibCore/Environment.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2021-2022, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2021-2022, Kenneth Myhra <kennethmyhra@serenityos.org>
+ * Copyright (c) 2021-2024, Sam Atkins <atkinssj@serenityos.org>
+ * Copyright (c) 2022, Matthias Zimmerman <matthias291999@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Environment.h"
+#include <AK/ByteString.h>
+
+#if defined(AK_OS_MACOS)
+#    include <crt_externs.h>
+#else
+extern char** environ;
+#endif
+
+namespace Core::Environment {
+
+char** raw_environ()
+{
+#if defined(AK_OS_MACOS)
+    return *_NSGetEnviron();
+#else
+    return environ;
+#endif
+}
+
+static Entry entry_from_string(char const* raw_input)
+{
+    StringView input { raw_input, strlen(raw_input) };
+    auto split_index = input.find('=');
+    if (!split_index.has_value()) {
+        return Entry {
+            .full_entry = input,
+            .name = input,
+            .value = ""sv,
+        };
+    }
+
+    return Entry {
+        .full_entry = input,
+        .name = input.substring_view(0, *split_index),
+        .value = input.substring_view(*split_index + 1),
+    };
+}
+
+void for_each_entry(Function<IterationDecision(Entry&)> callback)
```

Comment:
Not sure how idiomatic it is but range-based-for-loop-iteratable `Core::Environment::entries()` would be really cool. (I really don't like `Function`s where not necessary)

Reply:
+1 for begin()/end() instead, this is basically just a simple index range plus a mapper